### ISSUE TITLE
Kinship Representation Learning with Face Componential Relation

### ISSUE THREAD

**Issue by vitalwarley** [vitalwarley/research#51]
- Código: https://github.com/wtnthu/FaCoR
- Paper: https://arxiv.org/pdf/2304.04546.pdf

Chamou-me a atenção por estar entre os melhores resultados na comparação feita na tabela 6 de #50.

**Comment by vitalwarley** [comment]
### 1. Pre-Reading

- **Abstract Key Points:**

    - Problem

        - Research on [Kinship Recognition](https://app.capacities.io/56bb1d08-113e-4305-ad58-7b3e3eaf845e/0ded54ff-87ac-40ed-9994-78d0680b6ba8) mostly focuses on heuristic designs without considering the spatial correlation between face images. #KinshipVerification #KinshipRecognition#Research

            - What are "heuristic designs" as mentioned in the abstract?

    - Method

        - **Fa**ce **Co**mponential **R**elation **Net**work (FaCoRNet).

            - What does it do? They aim to learn discriminative kinship representations embedded with the relation information between face components.

            - How does it do? 

                - Cross-attention mechanism: learn important facial regions from face components among images.

                - Relation-Guided Contrastive Learning: adapt the loss function by the guidance from cross-attention to learn more discriminative feature representations.

    - Results

        - Outperform previous SOTA by largin margins.

- **Figures/Tables Noted:**

    - Face Componential Relation Guidance

       ![image](https://github.com/vitalwarley/research/assets/6365065/481a319c-5baa-406f-91f4-f02e188d8dfa)

    - Face Componential Relation Network
    
       ![image](https://github.com/vitalwarley/research/assets/6365065/1cf9238b-363f-421c-b914-b0806d5b745b)

    - SOTA on FIW

       ![image](https://github.com/vitalwarley/research/assets/6365065/b78e434e-5831-4944-b3ea-14c937851001)


        - What "Contrastive (naive)" means?

        [Qual a diferença entre os backbones ArcFace e AdaFace?](https://app.capacities.io/56bb1d08-113e-4305-ad58-7b3e3eaf845e/71f90588-7912-4354-b4bc-244578ebb694)

    - Performance by quality-filtered protocol
    
       ![image](https://github.com/vitalwarley/research/assets/6365065/0eb1d064-62ec-44ba-a087-75c29cedf9bf)


    - Component Analysis 

       ![image](https://github.com/vitalwarley/research/assets/6365065/95e761dc-f337-4116-94f7-08d4cae96242)

        - How can we interpret these improvements? Rel-Guide shows only 0.003, 0.003, and 0.039 improvement.

    - Visual analyis with t-SNE
   
       ![image](https://github.com/vitalwarley/research/assets/6365065/670b5227-15f6-4702-b6c8-c6eca26c7ef4)

        Insight: see the baby from family 22. What if we manage to cluster individuals by age, then by family? The baby would be proximal to other babies, not adults.

        I am thinking about the notion of [Recognizability Index](https://app.capacities.io/56bb1d08-113e-4305-ad58-7b3e3eaf845e/70fafa69-e1d2-46d1-abe8-5c79d7a85540). In kinship we wouldn't have Unrecognizable Identities, but Unverifiable Individuals. A baby is somewhat unverifiable.

        But if we cluster by age, we could complicate the kinship verification, as some kinship types rely on it. It would be useful to know how many errors on present SOTA are due to age. Gender, also, sure.

        What if I pre-train the model from [Recognizability Embedding Enhancement for Very Low-Resolution Face Recognition and Quality Estimation](https://app.capacities.io/56bb1d08-113e-4305-ad58-7b3e3eaf845e/5a24ea28-181f-4dbb-a60e-67db181e2900), which proposed the [Recognizability Index](https://app.capacities.io/56bb1d08-113e-4305-ad58-7b3e3eaf845e/70fafa69-e1d2-46d1-abe8-5c79d7a85540), on the family classification task à lá [Achieving Better Kinship Recognition Through Better Baseline](https://app.capacities.io/56bb1d08-113e-4305-ad58-7b3e3eaf845e/c5272d32-2f4f-45db-89e5-39f8085bac0b)? Unrecognizable individuals would cluster? A baby, for example, is a good example of potential unrecognizable individual in the family context.

        The RI aims to improve recognition of low resolution images. This paper here aims, as future work, to incorporate face quality scores into the training process. There is something to do here.

    - Cross-Attention map results

       ![image](https://github.com/vitalwarley/research/assets/6365065/2e72e535-cfe3-4d97-9f35-1d9d499b9641)

- Conclusion Key Points

    - Proposal

        - FaCoRNet is an attention-based model designed for learning correlation between image pairs in terms of face components. Its FaCoR module achieves adaptive learning correlation between image pairs and learns important face components for kinship recognition. It helps against large variations in facial appearance. Cross-attention estimation as a relation indicator guides the regular contrastive loss without the need for heuristic tuning.

            - What heuristic tuning is?

    - Future work

        - Incorporate face quality scores into the training process, aiming to mitigate the issues from low-quality face images.

        - Multi-modal information (e.g., text, metadata) to compensate for the vision-based methods.
